app/src/main/java/com/hedvig/app/util/JSONObject.kt

```diff
@@ -48,3 +53,11 @@ operator fun JSONObject.plus(other: JSONObject): JSONObject {
     }
     return clone
 }
+
+fun JSONArray.toStringArray(): List<String> {
+    val list: MutableList<String> = ArrayList()
+    for (i in 0 until length()) {
+        list.add(get(i).toString())
+    }
+    return list
+}
```

Comment:
```suggestion
fun JSONArray.toStringArray() = map { it.toString() }
```

Reply:
Cant seem to iterate over a JSONArray 🤔

Reply:
Oh damn, my bad. Maybe it could be worth implementing an extension that provides an iterator for `JSONArray`? With that, we could make a lot more types of iterations on them 🙂

Reply:
Yeah we could do something like
```
fun <T> JSONArray.toArray(action: () -> T): List<T> {
    val list: MutableList<T> = ArrayList()
    for (i in 0 until length()) {
        list.add(action())
    }
    return list
}
```

But I think we can tackle that problem in case we need it?

Reply:
Personally, I'd implement a `JSONArrayIterator`, we have a few examples of custom iterators in the project. (lint now tells me that they are incorrectly implemented, so we have that to deal with, but whatever)

Having iterators available for this is generally the correct level of abstraction to create, in my opinion, but if you feel that it's entirely unnecessary, we can skip it. 🙂